projects/packages/sync/src/class-queue.php

```diff
@@ -46,6 +78,180 @@ public function __construct( $id ) {
 		$this->id           = str_replace( '-', '_', $id ); // Necessary to ensure we don't have ID collisions in the SQL.
 		$this->row_iterator = 0;
 		$this->random_int   = wp_rand( 1, 1000000 );
+
+		/**
+		 * If the Dedicated table is not force-disabled (status = -1), we can try to initialize it.
+		 */
+		if ( ! self::dedicated_table_disabled() ) {
+			$this->maybe_initialize_dedicated_sync_table();
+			$this->queue_storage = new Queue_Storage_Table( $this->id );
```

Comment:
I believe we need to only set this if `$this->maybe_initialize_dedicated_sync_table();` returns `true`